File: 07-week-6--foundational-data-structures/03-bonus-algorithm--recursive-string-reverse/javascript/reverse_string.js
function reverseString(str) {
  if (str.length === 0) {
    return "";
  }

  return str.slice(-1) + reverseString(str.slice(0, -1));
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("ih"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  reversed = ""
  i = str.length - 1
  while i >= 0 
    reversed += str[i]
    i = i - 1
  end
  p reversed == str
end

reverse_string("racecar")

# if __FILE__ == $PROGRAM_NAME
#   puts "Expecting: 'ih'"
#   puts "=>", reverse_string('hi')

#   puts

#   puts "Expecting: 'ybabtac'"
#   puts "=>", reverse_string('catbaby')

#   # Don't forget to add your own!
# end

# Please add your pseudocode to this file
# And a written explanation of your solution